model/task_history.go

```diff
@@ -351,16 +354,9 @@ func (iter *taskHistoryIterator) GetChunk(v *Version, numBefore, numAfter int, i
 	return chunk, nil
 }
 
-func (self *taskHistoryIterator) GetDistinctTestNames(numCommits int) ([]string, error) {
-	session, mdb, err := db.GetGlobalSessionFactory().GetSession()
-	if err != nil {
-		return nil, errors.Wrap(err, "problem getting database session")
-	}
-	defer session.Close()
-
-	session.SetSocketTimeout(time.Minute)
-
-	pipeline := mdb.C(task.Collection).Pipe(
+func (self *taskHistoryIterator) GetDistinctTestNames(env evergreen.Environment, ctx context.Context, numCommits int) ([]string, error) {
```

Comment:
I don't know why, but there's a thing to [put the ctx first](https://golang.org/pkg/context/#pkg-overview:~:text=Do%20not%20store%20Contexts%20inside%20a,the%20first%20parameter%2C%20typically%20named%20ctx%3A)

Reply:
env > context, alas